CiscoESA/cisco-esa/ingest/parser.yml

```
@@ -0,0 +1,313 @@
+name: cisco-esa
+pipeline:
+
+  - name: parsed_event
+    external:
+      name: cef.parse-cef
+
+  - name: parse_start_date
+    filter: "{{parsed_event.message.start != null or parsed_event.message.startTime != null}}"
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{parsed_event.message.start or parsed_event.message.startTime}}"
+        output_field: date
+
+  - name: parse_end_date
+    filter: "{{parsed_event.message.end != null or parsed_event.message.endTime != null}}"
+    external:
+      name: date.parse
+      properties:
+        input_field:  "{{parsed_event.message.end or parsed_event.message.endTime}}"
+        output_field: date
+
+  # Parse 'rt' (as a timestamp or as a date)
+  - name: parse_receipt_time
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{parsed_event.message.rt}}"
+        output_field: date
+
+  - name: set_timestamp
+  - name: set_cef_header_fields
+  - name: set_cef_extension_fields
+  - name: set_cef_extension_fields_cef
+
+stages:
+  set_timestamp:
+
+    actions:
+      # Default timestamp is the start or startTime field
+      - set:
+          '@timestamp': "{{parse_start_date.date}}"
+
+      # Second choice is the receipt time (rc) field
+      - set:
+          '@timestamp': "{{parse_receipt_time.date}}"
+        filter: "{{parse_start_date == null and parse_receipt_time != null}}"
+
+      # Last choice is the end or endTime field
+      - set:
+          '@timestamp': "{{parse_end_date.date}}"
+        filter: "{{parse_start_date == null and parse_receipt_time == null and parse_end_date != null}}"
+
+  set_cef_header_fields:
+  # CEF:Version|Device Vendor|Device Product|Device Version|Device Event Class ID|Name|Severity|[Extension]
+    actions:
+
+      - set:
+          observer.vendor: '{{parsed_event.message.DeviceVendor}}'
+          observer.type: '{{parsed_event.message.DeviceProduct}}'
+          observer.version: '{{parsed_event.message.DeviceVersion}}'
+          rule.id: '{{parsed_event.message.DeviceEventClassID}}'
+
+      - set:
+          # Match only for agentSeverity of type int
+          event.severity: '{{parsed_event.message.Severity}}'
+        filter: "{{parsed_event.message.Severity | re_match('[1-9]|10')}}"
+
+
+  set_cef_extension_fields:
+    actions:
+
+      - set:
+          event.action: '{{parsed_event.message.act}}'
+          network.protocol: '{{parsed_event.message.app}}'
+          event.reason: '{{parsed_event.message.msg}}'
+          file.name: '{{parsed_event.message.fname}}'
+          file.size: '{{parsed_event.message.fsize}}'
+          source.ip: '{{parsed_event.message.src or parsed_event.message.sourceAddress}}'
+          source.user.name: '{{parsed_event.message.suser}}'
```

It seems that [`source.user.email`](https://www.elastic.co/guide/en/ecs/8.6/ecs-user.html#field-user-email) is more suitable
```suggestion
          source.user.email: '{{parsed_event.message.suser}}'
```
I think we can use a grok to extract the `source.user.name` and the `source.user.domain` from the email